File: pkg/ingestion/testdata/python/class_methods.py
"""Sample class with methods."""

class UserService:
    """Service for managing users."""

    def __init__(self, db):
        self.db = db

    def get_user(self, user_id: int):
        """Get user by ID."""
        return self.db.query(user_id)

    def create_user(self, name: str):
        """Create a new user."""
        return self.db.insert(name)

File: pkg/ingestion/testdata/typescript/class_methods.ts
// Class with methods
export class UserService {
    private db: any;

    constructor(db: any) {
        this.db = db;
    }

    getUser(userId: number): any {
        return this.db.query(userId);
    }

    createUser(name: string): any {
        return this.db.insert(name);
    }
}
